### PR TITLE
Update for Sensu Go Enablement

### PR DESCRIPTION
**Pull request by jspaleta**
## Pull Request Checklist

**Is this in reference to an existing issue?**
Closes issue #3  
Closes PRs #10 #8 #7 #5 #4   

#### General

- [X] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [X] RuboCop passes

#### Purpose
1. Update gem deps
2. Changes to make it possible to build automated testing without having to send post to twitter.

#### Known Compatability Issues
None

#### Local tests:
Using `/etc/sensu/conf.d/twitter.json` from README.md

Sensu Core 1.x Test:
```
echo '{"client":{"subscriptions":["twitter"]} }' | ./bin/handler-twitter.rb  --verbose --disable-update
```
Output:
```
Settings: {"myservice-status"=>{"consumer_key"=>"123", "consumer_secret"=>"123", "oauth_token"=>"123", "oauth_token_secret"=>"123", "sensusub"=>"twitter"}}
Event: {"client"=>{"subscriptions"=>["twitter"]}, "occurrences"=>1, "check"=>{}}
Account: myservice-status
  Matching Subscription: twitter
  Consumer Key: 123
  Consumer Secret: 123
  Oauth Token: 123
  Oauth Token Secret: 123
Twitter Client: #<Twitter::REST::Client:0x000055d5256a5220>
  Consumer Key: 123
  Consumer Secret: 123
  Access Token: 123
  Access Token Secret: 123
ALERT - unknown/unknown:  Time: 2018-12-24 16:26:10 -0900 
```
Sensu Go Test:
```
echo '{"entity":{"subscriptions":["twitter"]} }' | ./bin/handler-twitter.rb  --verbose --disable_update --map-go-event-into-ruby
```
Output
```
Settings: {"myservice-status"=>{"consumer_key"=>"123", "consumer_secret"=>"123", "oauth_token"=>"123", "oauth_token_secret"=>"123", "sensusub"=>"twitter"}}
Event: {"entity"=>{"subscriptions"=>["twitter"], "subscribers"=>["twitter"]}, "occurrences"=>1, "check"=>{"subscribers"=>nil, "source"=>nil}, "client"=>{"subscriptions"=>["twitter"], "subscribers"=>["twitter"]}, "action"=>"unknown::go_event", "go_event_mapped_into_ruby"=>true}

Account: myservice-status
  Matching Subscription: twitter
  Consumer Key: 123
  Consumer Secret: 123
  Oauth Token: 123
  Oauth Token Secret: 123
Twitter Client: #<Twitter::REST::Client:0x0000561fa2904c60>
  Consumer Key: 123
  Consumer Secret: 123
  Access Token: 123
  Access Token Secret: 123
ALERT - unknown/unknown:  Time: 2018-12-24 16:28:05 -0900 
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,29 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## Unreleased
+### Added 
+- Bump sensu-plugin runtime dep to 3.0 with Sensu Go to Sensu Core 1.x migration option (@jspaleta)
+- Added 2.5.1 ruby support (@jspaleta)
+- Added --verbose cmdline switch to aid in local testing
+- Added --disable_update cmdline switch to aid in local testing
+
+### Breaking Changes
```
Comment (majormoses) [inline CHANGELOG.md:13]: I personally order `### Breaking Changes` before other sections (other than security) as it should be the first thing someone evaluates for upgrade.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,29 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+- Dropped 2.0 ruby support (@jspaleta)
```
Comment (majormoses) [inline CHANGELOG.md:14]: really we are dropping for ruby versons `> 2 && <= 2.3` for simplicity sake I ussually say drop support for ruby `< 2.3` but thats a matter of preference I think.

**Comment by jspaleta** [comment]
The travis build failures are from the old ruby versions removed from travis.yml in this commit.

File: .travis.yml [hunk 3]
```diff
@@ -27,9 +27,7 @@ deploy:
   on:
     tags: true
     all_branches: true
-    rvm: 2.0
```
Comment (majormoses) [inline .travis.yml:30]: you also need to remove these versions from the rvm array as well.

**Review by majormoses — APPROVED** [review]
LGTM

You can self merge on CI pass.

**Pull request merged**